core/unit_test/TestViewCtorProp.hpp

```diff
@@ -0,0 +1,126 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+
+namespace {
+
+// Check Kokkos::Impl::is_view_label.
+TEST(TEST_CATEGORY, is_view_label) {
+  static_assert(Kokkos::Impl::is_view_label<std::string>::value);
+
+  constexpr unsigned N = 3;
+  static_assert(Kokkos::Impl::is_view_label<const char[N]>::value);
+  static_assert(Kokkos::Impl::is_view_label<char[N]>::value);
+
+  // A char* is not a label. Thus, a label is distinguished from a pointer type.
+  static_assert(!Kokkos::Impl::is_view_label<char*>::value);
+}
+
+// Check traits of Kokkos::Impl::ViewCtorProp<>.
+TEST(TEST_CATEGORY, vcp_empty_traits) {
+  using vcp_empty_t = Kokkos::Impl::ViewCtorProp<>;
+
+  // Check that the empty view constructor properties class is default
+  // constructible. This is needed for calls of Kokkos::view_alloc().
+  static_assert(std::is_default_constructible_v<vcp_empty_t>);
+
+  static_assert(std::is_same_v<decltype(Kokkos::view_alloc()), vcp_empty_t>);
+}
+
+// Check traits of base class Kokkos::Impl::ViewCtorProp<void, std::string>.
+TEST(TEST_CATEGORY, vcp_label_base_traits) {
+  using vcp_label_base_t = Kokkos::Impl::ViewCtorProp<void, std::string>;
+
+  static_assert(std::is_same_v<typename vcp_label_base_t::type, std::string>);
+
+  // Check that the base class is default constructible. The default constructor
+  // may be called by the copy constructor of deriveded classes, such as when
+  // copy constructing a view constructor properties object from another view
+  // constructor properties object that holds fewer properties.
+  static_assert(std::is_default_constructible_v<vcp_label_base_t>);
```

Comment:
@maartenarnst Please double-check that this is also the case on the develop branch. Otherwise, you'll need to add a constraint to your variadic constructor that you need at least on argument.

Reply:
Sorry, my mistake. It is already default constructible on the develop. Closing.